db/models/models.js

```diff
@@ -10,6 +10,88 @@ const fetchTopics = () => {
     })
 }
 
+const fetchArticles = () => {
+    const sqlString = 
+    `
+        SELECT articles.*, COUNT(comments.comment_id) as comment_count
+        FROM articles
+        LEFT JOIN comments
+        ON articles.article_id = comments.article_id
+        GROUP BY articles.article_id
+        ORDER BY created_at DESC;
+    `
+    return db.query(sqlString)
+    .then(({rows}) => rows)
+}
+
+const fetchCommentsByArticleId = (article_id) => {
+    const sqlString = 
+    `
+        SELECT * FROM comments
+        WHERE article_id = $1
+        ORDER BY created_at DESC;
+    `
+    return db.query(sqlString, [article_id])
+    .then(({rowCount, rows}) => {
+        if (rowCount === 0) {
+            return Promise.reject({status: 404, message: 'Not Found'})
+        } else {
+            return rows
+        }
+    })
+}
+
+const addNewComment = (article_id, newCommentData) => {
+    if (Object.keys(newCommentData).length <= 1 || 
+        !newCommentData.username || 
+        !newCommentData.body
+    ) {
+        return Promise.reject({status: 400, message: 'Bad Request'})
+    }
+
+    const {username, body, votes = 0} = newCommentData
+
+    // checking if the username exists in the users database
+    const userString =  
+    `
+        SELECT * FROM users
+        WHERE username = $1;
+    
+    `
+    return db.query(userString, [username])
```

Comment:
this query wouldn't be needed, if you tried to add a user to the databse that didn't exist PSQL would throw an error